CoreScriptsRoot/Modules/Settings/Pages/GameSettings.lua

```diff
@@ -103,10 +103,38 @@ local function Initialize()
 
 		settingsDisabledInVR[this.FullscreenEnabler] = true
 
-		local fullScreenSelectionFrame = this.FullscreenEnabler.SliderFrame and this.FullscreenEnabler.SliderFrame or this.FullscreenEnabler.SelectorFrame
-
+		local Debounce = false
+		
 		this.FullscreenEnabler.IndexChanged:connect(function(newIndex)
-			GuiService:ToggleFullscreen()
+			if Debounce then return end
+			Debounce = true
+			if newIndex == 1 then
+				if not GameSettings:InFullScreen() then
+					GuiService:ToggleFullscreen()
+					this.FullscreenEnabler:SetSelectionIndex(1)
+				end
+			elseif newIndex == 2 then
+				if GameSettings:InFullScreen() then
+					GuiService:ToggleFullscreen()
+					this.FullscreenEnabler:SetSelectionIndex(2)
+				end
+			end
+			Debounce = false
+		end)
+
+		GameSettings.FullscreenChanged:connect(function(isFullScreen)
```

Comment:
Were you able to test these changes? I don't know if it works exactly like this, but you may be able to leave this.FullscreenEnabler.IndexChanged as it was before, and then just update the gui with this function (I believe ToggleFullscreen will fire this function)